addons/website_sale_autocomplete/static/src/js/addressForm.js

```diff
@@ -0,0 +1,88 @@
+        this.cityInput = document.querySelector('input[name="city"]');
+        this.zipInput = document.querySelector('input[name="zip"]');
+        this.countrySelect = document.querySelector('select[name="country_id"]');
+        this.stateSelect = document.querySelector('select[name="state_id"]');
+        this.dp = new DropPrevious();
+
+        this._onChangeStreet = debounce(this._onChangeStreet, 200);
+        this._super.apply(this, arguments);
+    },
+
+    _onChangeStreet: function(ev) {
+        if(!this.streetAndNumberInput.dataset.autocompleteEnabled) return;
+        this.dp.add(
+            this._rpc({
+                route: '/autocomplete/address',
+                params: {
+                    partial_address: ev.currentTarget.value
+                }
+            })).then((response) => {
+            if(response && response.template && response.template.trim()){
+                const inputContainer = ev.currentTarget.parentNode;
+                const dropdown = inputContainer.querySelector('.dropdown-menu');
+                if(dropdown) dropdown.remove();
+                inputContainer.appendChild($(response.template)[0])
+            }
+        })
+    },
+
+    _onClickAutocompleteResult: function(ev) {
+        const dropDown = ev.currentTarget.parentNode;
+
+        this._rpc({
+            route: '/autocomplete/address_full',
+            params: {
+                address: ev.currentTarget.innerText,
+                google_place_id: ev.currentTarget.dataset.googlePlaceId
+            }
+        }).then((result) => {
+            if(result.address){ // Don't change this field in case something goes wrong
+                if(result.address.formatted_street_number){
+                    this.streetAndNumberInput.value = result.address.formatted_street_number;
+                }
+                // Text fields, empty if no value in order to avoid the user missing old data.
+                this.zipInput.value = result.address.zip || '';
+                this.cityInput.value = result.address.city || '';
+
+                // Selects based on odoo ids
+                if(result.address.country){
+                    this.countrySelect.value = result.address.country;
+                    // Let the state select know that the country has changed so that he may fetch the correct states or disappear.
+                    this.countrySelect.dispatchEvent(new Event('change', {bubbles: true}));
+                }
+                if(result.address.state){
+                    // Waits for the stateSelect to update before setting the state.
+                    new MutationObserver((entries, observer) => {
+                        this.stateSelect.value = result.address.state;
+                        observer.disconnect();
+                    }).observe(this.stateSelect, {
+                        childList: true, // Trigger only if the options change
+                    });
```

Comment:
Since we trigger only if the options changes, will this work when changing to another address that uses the same country?
Example I input an address in Brussels and then the odoo farm, will it keep Brussels?
We can directly change `stateSelect`'s value when the country is kept the same.

Reply:
Actually, no, since the select is completely overwritten each time, the value we set immediately is also overwritten

Reply:
You don't have to override the value of the countrySelect when the value is the same either.